experimental/cache_analysis.py

```diff
@@ -0,0 +1,139 @@
+from matplotlib.pyplot import figure
+import matplotlib.pyplot as plt
+import io
+import numpy as np
+import pandas as pd
+from statistics import mean, median
+import collections
+import csv
+import seaborn as sns
+
+""" This functions draws two types of plots for cases where each x value has only one y value.
```

Comment:
The docstring goes inside the def, as so:

```py
def foo():
  """Description of what Foo does."""
  ...
```

Same for the rest of the functions.